### ISSUE TITLE
Gazebo-2.1.0 fails to compile on ARM

### ISSUE THREAD

**Issue by osrf-migration** [osrf/gazebo#1007]
**Original report ([archived issue](https://osrf-migration.github.io/gazebo-gh-pages/#!/osrf/gazebo/issues/1007)) by Jose Luis Rivero (Bitbucket: [Jose Luis Rivero](https://bitbucket.org/%7Bd12309b2-f745-42ee-b119-aec4fcdf81fe%7D/), GitHub: [j-rivero](https://github.com/j-rivero)).**

----------------------------------------
Fedora ARM building machines are reporting [failures while building gazebo-2.1.0 on an ARM platform](http://koji.fedoraproject.org/koji/taskinfo?taskID=6362362)

Thanks to @rmattes for the report and the debug, in his own words:

*
Basically, Qt typedefs "qreal" to double on every platform except for ARM.  On ARM it's a float, so you run into issues where people assume it's always a double, like in the templated implementations of std::min and std::max.
*
